clients/src/main/java/org/apache/kafka/common/security/kerberos/KerberosShortNamer.java

```diff
@@ -98,4 +98,9 @@ public String shortName(KerberosName kerberosName) throws IOException {
         throw new NoMatchingRule("No rules applied to " + toString());
```

Comment:
I think we should probably include information about the received `kerberosName` in the error message and use `principalToLocalRules.toString` instead of `toString`.